tests/cupy_tests/cuda_tests/test_memory.py

```diff
@@ -183,9 +183,69 @@ def test_memset(self):
             a_gpu += 1
 
 
+@testing.parameterize(*testing.product({
+    'use_streams': [True, False],
+}))
+@testing.gpu
+class TestMemoryPointerAsync(unittest.TestCase):
+
+    def setUp(self):
+        self.stream = stream_module.Stream() if self.use_streams else None
+
+    def test_copy_to_and_from_host_async(self):
+        a_gpu = memory.alloc(4)
+        a_cpu = ctypes.c_int(100)
+        a_gpu.copy_from_async(ctypes.cast(ctypes.byref(
+            a_cpu), ctypes.c_void_p), 4, stream=self.stream)
+
+        b_cpu = ctypes.c_int()
+        a_gpu.copy_to_host_async(
+            ctypes.cast(ctypes.byref(b_cpu), ctypes.c_void_p),
+            4, stream=self.stream)
+        assert b_cpu.value == a_cpu.value
+
+    def test_copy_from_device(self):
```

Comment:
Let's be consistent with other tests:
```suggestion
    def test_copy_from_device_async(self):
```